src/Databases/PostgreSQL/DatabaseMaterializedPostgreSQL.h

```diff
@@ -66,6 +70,13 @@ class DatabaseMaterializedPostgreSQL : public DatabaseAtomic
     void shutdown() override;
 
     String getPostgreSQLDatabaseName() const { return remote_database_name; }
+    
+    void detachTablePermanently(ContextPtr /*context*/, const String & /*name*/) override
+    {
+        throw Exception(ErrorCodes::NOT_IMPLEMENTED,
```

Comment:
So, if simple DETACH works like DETACH PERMANENTLY, then maybe we should throw exception on simple DETACH and remove table from replication on DETACH PERMANENTLY. What do you think?

Reply:
Makes sense... I'll do.